libs/net/dal/Services/BaseService`.cs

```diff
@@ -77,9 +77,25 @@ public virtual TEntity Update(TEntity entity)
 
     public virtual TEntity UpdateAndSave(TEntity entity)
     {
-        var result = Update(entity);
-        this.Context.CommitTransaction();
-        return result;
+        try
```

Comment:
This isn't a safe way to handle this.  A concurrency error means the process that is asking for the update has stale data.  They should be getting the error returned to them so that they figure out what to do.  They will need to get the current data, apply their changes, and then save again.